sdk/go/common/resource/plugin/plugin.go

```diff
@@ -554,14 +555,33 @@ func buildPluginArguments(opts pluginArgumentOptions) []string {
 	return args
 }
 
+func (p *plugin) healthCheck() bool {
+	if p.Conn == nil {
+		return false
+	}
+
+	// Check that the plugin looks alive by calling gRPC's Health Check service.
+	// Most plugins don't actually implement this service, which is OK as we treat
+	// an unimplemented status as OK.
+	health := grpc_health_v1.NewHealthClient(p.Conn)
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
```

Comment:
Does this mean that a crashed plugin will wait 3 seconds for a response?

Reply:
Yes, it's possible to take that long for crashed plugins. We can probably get by with a shorter amount of time, but I wasn't sure how much grace to give Idle connections to reconnect for plugins that are still alive. These are local connections, so reconnects _should_ be much faster than seconds. On my machine local connections take anywhere from 216.917µs to 339.833µs. 500ms (half a second) should be plenty, I can try reducing it to that.